chain-signatures/node/src/protocol/signature.rs

```diff
@@ -194,14 +241,21 @@ impl SignQueue {
             let request = self.organize_request(threshold, &stable, indexed, false);
             let is_mine = request.proposer == self.me;
             if is_mine {
-                self.my_requests.push_back(request);
+                self.my_requests.push_back(sign_id);
                 crate::metrics::NUM_SIGN_REQUESTS_MINE
                     .with_label_values(&[my_account_id.as_str()])
                     .inc();
-            } else {
-                self.other_requests
-                    .insert(request.indexed.id.clone(), request);
             }
+            if let Some(pending) = self.pending.remove(&sign_id) {
+                if pending.send(request.clone()).is_err() {
+                    tracing::warn!(
+                        ?sign_id,
+                        "pending sign request channel closed before able to send request"
+                    );
+                }
+            }
+
```

Comment:
Is it possible under any circumstances that an incoming request can't be matched? Maybe bad timing or  maybe a duplicated incoming request?

Reply:
so, this should be sequential right now since only `SignatureSpawner` organizes requests. When requests come in, this is the only point where they would get organized. For duplicate requests, a pending request would never get created since the request is already in the system and `get_or_pending` would return `PendingRequest::Available(request)` here.

Maybe it's less clear right now how the flow of requests goes but that just means we need to do some refactoring to make it clearer